admission_control_overload_test.py

```diff
@@ -0,0 +1,91 @@
+
+    def run_load_and_check_for_admission_control(self, prepare_base_cmd, stress_base_cmd, job_num=16):  # pylint: disable=invalid-name
+        # pylint: disable=too-many-locals
+
+        prometheus = PrometheusDBStats(self.monitors.nodes[0].public_ip_address)
+        transport_requests_memory_available = 'scylla_transport_requests_memory_available'  # pylint: disable=invalid-name
+        node_procs_blocked = 'scylla_transport_requests_blocked_memory'
+
+        # run a write workload
+        if prepare_base_cmd:
+            prepare_stress_queue = self.run_stress_thread(stress_cmd=prepare_base_cmd, stress_num=2, prefix='preload-',
+                                                          stats_aggregate_cmds=False)
+            # self.verify_stress_thread(cs_thread_pool=stress_queue)
+            self.get_stress_results(prepare_stress_queue)
+
+            time.sleep(10)
+
+        # run a read workload
+        stress_queue = []
+        stress_res = []
+        stress_queue.append(self.run_stress_thread(stress_cmd=stress_base_cmd, stress_num=job_num,
+                                                   stats_aggregate_cmds=False))
+
+        start_time = time.time()
+        max_transport_requests_memory_available = 0  # pylint: disable=invalid-name
+        is_admission_control_triggered = False
+        while stress_queue:
+            stress_res.append(stress_queue.pop(0))
+            while not all(future.done() for future in stress_res[-1].results_futures):
+                now = time.time()
+                transport_res = prometheus.query(transport_requests_memory_available, start_time, now)
+                node_procs_res = prometheus.query(node_procs_blocked, start_time, now)
+
+                sum_values = 0
+                for shard in transport_res:
+                    sum_values += int(shard['values'][0][1])
+                max_transport_requests_memory_available = max(  # pylint: disable=invalid-name
+                    sum_values, max_transport_requests_memory_available)
+                if sum_values < max_transport_requests_memory_available * 0.3:
+                    max_transport_requests_memory_available = sum_values  # pylint: disable=invalid-name
+                    for node in node_procs_res:
+                        if node['values'][0][1] > 0:
+                            self.log.info('Admission control was triggered')
+                            is_admission_control_triggered = True
+                start_time = now
+                time.sleep(10)
+
+        results = []
+        for stress in stress_res:
+            results.append(self.get_stress_results(stress))
+        # results = self.verify_stress_thread(cs_thread_pool=stress_queue)
+
+        self.verify_no_drops_and_errors(starting_from=self.start_time)
+        # TODO: raise exception if is_admission_control_triggered is false
+        if not is_admission_control_triggered:
+            raise Exception('Admission Control wasn\'t triggered')
```

Comment:
split this function into several logically separated functions.

Reply:
done